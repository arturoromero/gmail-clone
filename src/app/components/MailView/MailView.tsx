import React from "react";
import { makeStyles, Box, IconButton, Button } from "@material-ui/core";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import KeyboardArrowLeftIcon from "@material-ui/icons/KeyboardArrowLeft";
import KeyboardArrowRightIcon from "@material-ui/icons/KeyboardArrowRight";
import DeleteIcon from "@material-ui/icons/Delete";
import ArrowBackIcon from "@material-ui/icons/ArrowBack";
import ArchiveIcon from "@material-ui/icons/Archive";
import ReportIcon from "@material-ui/icons/Report";
import MarkunreadIcon from "@material-ui/icons/Markunread";
import WatchLaterIcon from "@material-ui/icons/WatchLater";
import CheckCircleIcon from "@material-ui/icons/CheckCircle";
import FolderIcon from "@material-ui/icons/Folder";
import LabelIcon from "@material-ui/icons/Label";
import mailListStyles from "./mailViewStyles";
import { useSelector } from "react-redux";
import { useLocation } from "react-router-dom";
import { history } from "../../../App";

const useStyles = makeStyles(mailListStyles);

interface IMailView {}

const MailView: React.FC<IMailView> = (props) => {
  const { search } = useLocation();
  const urlParams = new URLSearchParams(search);
  const currentMail = urlParams.getAll("mail");
  const { getMailList } = useSelector((state) => ({
    getMailList: state,
  }));
  const mailList: any = getMailList;
  const mailItems: [] = mailList.mailList.data;
  const goBack = () => {
    history.goBack();
  };
  const classes = useStyles();
  return (
    <Box>
      <Box className={classes.mailListHeader}>
        <Box className={classes.mailListHeaderLeft}>
          <Box className={classes.iconGroup}>
            <IconButton onClick={goBack}>
              <ArrowBackIcon />
            </IconButton>
            <IconButton>
              <ArchiveIcon />
            </IconButton>
            <IconButton>
              <ReportIcon />
            </IconButton>
            <IconButton>
              <DeleteIcon />
            </IconButton>
          </Box>
          <Box className={classes.iconGroup}>
            <IconButton>
              <MarkunreadIcon />
            </IconButton>
            <IconButton>
              <WatchLaterIcon />
            </IconButton>
            <IconButton>
              <CheckCircleIcon />
            </IconButton>
          </Box>
          <Box className={classes.iconGroup}>
            <IconButton>
              <FolderIcon />
            </IconButton>
            <IconButton>
              <LabelIcon />
            </IconButton>
            <IconButton>
              <MoreVertIcon />
            </IconButton>
          </Box>
        </Box>
        <Box className={classes.mailListHeaderRight}>
          <Button className={classes.paginationText}>1 of 1</Button>
          <Box className={classes.paginationIconsWrapper}>
            <KeyboardArrowLeftIcon />
            {"  "} <KeyboardArrowRightIcon />
          </Box>
        </Box>
      </Box>
      <Box className={classes.mailBody}>
        {mailItems.map((item: any, index) =>
          item.id === currentMail.toString() ? (
            <Box key={item.id}>
              <Box className={classes.mailBodySubject}>{item.subject}</Box>
              <Box className={classes.mailSender}>
                <Box>
                  <img
                    alt="avatar"
                    src="https://i.pravatar.cc/100"
                    className={classes.randomAvatar}
                  />
                </Box>
                <Box className={classes.mailSenderText}>{item.sender}</Box>
              </Box>
              <Box
                dangerouslySetInnerHTML={{
                  __html: item.body,
                }}
              />
            </Box>
          ) : (
            ""
          )
        )}
      </Box>
    </Box>
  );
};

export default MailView;
