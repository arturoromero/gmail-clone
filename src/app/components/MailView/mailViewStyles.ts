const mailViewStyles: any = {
    mailViewWrapper: {
        display: 'flex',
        width:'100%',
    },
    mailListWrapper: {
        display: 'flex',
        width:'100%',
    },
    mailListHeader: {
        display: 'flex',
        position: 'sticky',
        top: '0',
        backgroundColor: '#fff',
        width: '100%',
        height: '48px',
        boxShadow: '0 3px 2px -2px rgb(0 0 0 / 10%)',
        justifyContent: 'space-between',
        zIndex: '9999',
    },
    mailListHeaderLeft: {
        display: 'flex',
        alignItems: 'center',
    },
    mailListHeaderRight: {
        fontSize: '.75rem',
        color: '#5f6368',
        display: 'flex',
        paddingRight: '8px',
    },
    paginationText: {
        padding: '6px 8px',
        fontSize: '.75rem',
        fontWeight: '400',
        color: '#616569',
        textTransform: 'none',
        display: 'flex',
        margin: '7px 0',
        borderRadius: '8px',
        '&:hover':{
            backgroundColor: '#e5e5e5'
        }
    },
    paginationIconsWrapper: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    mailListContent: {
        overflowY: 'auto',
        height: '100vh',
    },
    iconGroup:{
        display: 'flex',
        margin: '0 7px',
        boxShadow: 'inset -1px 0 0 rgb(100 121 143 / 12%)',
        height: '24px',
        alignItems: 'center',
        '& button': {
            height: '46px',
        },
        '&:last-child': {
            boxShadow: 'none',
        }
    },
    mailBody: {
        padding: '26px',
        color: '#555',
        font: 'small/ 1.5  Arial,Helvetica,sans-serif ',
    },
    mailBodySubject: {
        fontSize: '1.375rem',
        color: '#202124'
    },
    randomAvatar: {
        width: '40px',
        height: '40px',
        marginTop: '12px',
        borderRadius: '20px',
    },
    mailSender: {
        display: 'flex',
        alignItems: 'center',
    },
    mailSenderText: {
        fontWeight: '600',
        color: '#202124',
        margin: '8px 0 0 8px',
    }


    
};

export default mailViewStyles;
