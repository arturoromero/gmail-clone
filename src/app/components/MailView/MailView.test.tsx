import React from "react";
import renderer from "react-test-renderer";
import MailView from "./MailView";

describe("MAIL TEMPLATE COMPONENT", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<MailView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
