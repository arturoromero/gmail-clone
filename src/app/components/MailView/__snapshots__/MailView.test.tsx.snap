// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MAIL TEMPLATE COMPONENT renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 makeStyles-mailViewWrapper-1"
>
   HERE LIES A MAIL 
</div>
`;
